jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -418,11 +418,24 @@ public abstract boolean renameTo(Resource dest)
     /**
      * Returns the resource contained inside the current resource with the
      * given name, which may or may not exist.
+     * <p>
+     *     Constraints:
+     *     <ul>
+     *         <li>The path is expected to be in URI path-like format.</li>
```

Comment:
This "URI path-like" format is the ambiguous decoded path space that is returned from `getPathInfo` etc.
I think it is this space that needs to be better defined, which is part of the conversation at: https://github.com/eclipse-ee4j/servlet-api/issues/18#issuecomment-879769757

The path is "URI-like" in that it is `/` separated segments, but it is not a URI as it cannot contain encoded characters, specifically it cannot contain an encoded `/`, so there are some paths that cannot be specified.